hooks/index.ts

```diff
@@ -21,3 +27,37 @@ export async function caseStudyFullInfo() {
   const caseStudies: CaseStudyType[] = await getCaseStudy();
   return caseStudies;
 }
+
+export async function getHomePageData() {
+  try {
+    const caseStudies = await caseStudyInfoHomepage();
+    const testimonials = await getTestimonials();
+    const workExperienceList = await getWorkExperiences();
+    const services = await getServicesProvided();
+    const mySkills = await getMySkills();
```

Comment:
You are doing great. There are a couple of things we want to consider here. 

1. Using Promise.all would allow these asynchronous operations to execute concurrently, meaning they all get initiated almost at the same time, and the function will wait for all to complete. This can generally lead to faster load times as well. 
An example to consider:
```
 try {
    const results = await Promise.all([
      caseStudyInfoHomepage(),
      getTestimonials(),
      getWorkExperiences(),
      getServicesProvided(),
      getMySkills(),
    ]);

    const [caseStudies, testimonials, workExperienceList, services, mySkills] = results;
    ...
    ...
```
2. Error Handling: The way it is currently set up, if one API call fails, none of the data gets returned. Consider handling errors for individual requests so that the failure of one doesn't impact the others.

3. Console.error: The error is logged to the console, but you should return something more meaningful. 
`return { error: error.message || "An unknown error occurred." };`

Reply:
Hi @cdsalty Thanks a lot I think I've got this working a lot better now. I'm using Promise.allSettled which is also letting me go through each operation and even if one fails the others will still fire and return something if it is able to